inference-engine/include/ie_common.h

```diff
@@ -274,73 +281,210 @@ struct QueryNetworkResult {
     ResponseDesc resp;
 };
 
-/** @brief This class represents StatusCode::GENERIC_ERROR exception */
-class GeneralError : public std::logic_error {
-    using std::logic_error::logic_error;
+namespace details {
+struct INFERENCE_ENGINE_DEPRECATED("Use InferRequest::Exception")
+INFERENCE_ENGINE_API_CLASS(InferenceEngineException) : public std::runtime_error {
+    using std::runtime_error::runtime_error;
+    bool hasStatus() const {return true;}
+    StatusCode getStatus() const;
 };
+}  // namespace details
 
-/** @brief This class represents StatusCode::NOT_IMPLEMENTED exception */
-class NotImplemented : public std::logic_error {
-    using std::logic_error::logic_error;
+/**
+ * @brief Base Inference Engine exception class
+ */
+IE_SUPPRESS_DEPRECATED_START
+struct INFERENCE_ENGINE_API_CLASS(Exception) : public details::InferenceEngineException {
+    using InferenceEngineException::InferenceEngineException;
```

Comment:
I wonder how RTTI will work here since we don't export any function from IE (application / plugin / IE will have their own copies of these exception clang usually fails to cast / dynamic cast them)

Maybe we need to export dtors at least as it was done for old exception?